packages/react-network/src/components/NetworkEffect.tsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import {Effect} from '@shopify/react-effect';
+
+import {Consumer} from '../context';
+import {Manager} from '../Manager';
+
+interface Props {
+  perform(manager: Manager): void;
+}
+
+export const EFFECT_ID = Symbol('network');
+
+export default function NetworkEffect({perform}: Props) {
+  if (typeof window !== 'undefined') {
```

Comment:
Is there a more explicit server check we can do? I worry about this breaking for example in test environments where the DOM gets mocked.

Reply:
Nope, this is the best we have. This is actually kind of a feature for tests, since we generally don't really care that any of these will do anything real in tests (it's basically built-in mocking). Jest does allow per-file setting of the environment though so it is easy enough to verify these as they would appear in Node.